src/dictionary/flat_file/flat_file.c

```diff
@@ -182,12 +218,15 @@ flat_file_scan(
 			prev_offset = cur_offset;
 		}
 
-		size_t i;
+		flat_file->current_loaded_region	= (prev_offset - flat_file->start_of_data) / flat_file->row_size;
+		flat_file->num_in_buffer			= num_records_to_process;
+
+		int32_t i;
 
-		for (i = 0; i < num_records_to_process; i++) {
+		for (i = scan_forwards ? 0 : num_records_to_process - 1; scan_forwards ? (size_t) i < num_records_to_process : i >= 0; scan_forwards ? i++ : i--) {
```

Comment:
Change scan to take a macro define (i.e. SCAN_FOWARDS, SCAN_BACKWARDS) instead of a boolean